libs/auth/src/memory_card.ts

```diff
@@ -141,11 +145,43 @@ export class MemoryCard implements Card {
           cardSummary.status === 'error' ? 'card_error' : cardSummary.status,
       };
     }
-    const { user } = parseUserDataFromCardSummary(cardSummary);
-    return {
-      status: 'ready',
-      cardDetails: user ? { jurisdiction: DEV_JURISDICTION, user } : undefined,
-    };
+    const { user, pin } = parseUserDataFromCardSummary(cardSummary);
+    if (!user) {
+      return {
+        status: 'ready',
+        cardDetails: undefined,
+      };
+    }
+    let cardDetails: CardDetails;
+    switch (user.role) {
+      case 'system_administrator': {
+        cardDetails = {
+          jurisdiction: DEV_JURISDICTION,
+          user,
+        };
+        break;
+      }
+      case 'election_manager': {
+        cardDetails = {
+          jurisdiction: DEV_JURISDICTION,
+          user,
+        };
+        break;
+      }
+      case 'poll_worker': {
+        cardDetails = {
+          jurisdiction: DEV_JURISDICTION,
+          user,
+          hasPin: pin !== undefined,
```

Comment:
Was nearly going to punt on updating this deprecated memory card code path but had to do more than half the work just to get the TS compiler happy. So went ahead and completed things